From 5e769cde3c41f466db1477882e8b22686bb1617c Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Wed, 19 Nov 2014 16:14:39 -0500
Subject: [ethernet] r8169: add support more chips to get mac address from backup mac address register

Message-id: <1416413687-12219-9-git-send-email-ivecera@redhat.com>
Patchwork-id: 100356
O-Subject: [RHEL7.1 PATCH 08/16] r8169:add support more chips to get mac address from backup mac address register
Bugzilla: 1165764
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

BZ: #1165764

Upstream commit(s):
commit 89cceb2729c752e6ff9b3bc8650a70f29884f116
Author: Chun-Hao Lin <hau@realtek.com>
Date:   Wed Oct 1 23:17:15 2014 +0800

    r8169:add support more chips to get mac address from backup mac address register

    RTL8168FB RTL8168G RTL8168GU RTL8411 RTL8411B RTL8106EUS RTL8402 can
    support get mac address from backup mac address register.

    Signed-off-by: Chun-Hao Lin <hau@realtek.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/realtek/r8169.c b/drivers/net/ethernet/realtek/r8169.c
index 49967c6..51abd9d 100644
--- a/drivers/net/ethernet/realtek/r8169.c
+++ b/drivers/net/ethernet/realtek/r8169.c
@@ -7704,7 +7704,16 @@ static int rtl_init_one(struct pci_dev *pdev, const struct pci_device_id *ent)
  u64_stats_init(&tp->tx_stats.syncp);
 
  /* Get MAC address */
- if (tp->mac_version == RTL_GIGA_MAC_VER_45 ||
+ if (tp->mac_version == RTL_GIGA_MAC_VER_35 ||
+     tp->mac_version == RTL_GIGA_MAC_VER_36 ||
+     tp->mac_version == RTL_GIGA_MAC_VER_37 ||
+     tp->mac_version == RTL_GIGA_MAC_VER_38 ||
+     tp->mac_version == RTL_GIGA_MAC_VER_40 ||
+     tp->mac_version == RTL_GIGA_MAC_VER_41 ||
+     tp->mac_version == RTL_GIGA_MAC_VER_42 ||
+     tp->mac_version == RTL_GIGA_MAC_VER_43 ||
+     tp->mac_version == RTL_GIGA_MAC_VER_44 ||
+     tp->mac_version == RTL_GIGA_MAC_VER_45 ||
      tp->mac_version == RTL_GIGA_MAC_VER_46 ||
      tp->mac_version == RTL_GIGA_MAC_VER_47 ||
      tp->mac_version == RTL_GIGA_MAC_VER_48) {
-- 
1.7.1